src/Core/Services/OrganizationService.cs

```diff
@@ -51,5 +56,20 @@ public async Task ClearAllAsync(string userId)
         {
             await _stateService.SetOrganizationsAsync(null, userId);
         }
+
+        public async Task<string> GetClaimedOrganizationDomainAsync(string userEmail)
+        {
+            try
+            {
+                var response = await _apiService.GetOrgDomainSsoDetailsAsync(userEmail);
+                return response.OrganizationIdentifier;
+            }
+            catch (System.Exception ex)
+            {
+                // api throws 404 if there is no domain claimed
+                _logger.Value.Exception(ex);
+                return string.Empty;
+            }
+        }
```

Comment:
if the `userEmail` is `null`, does it makes sense to call the service? I think we could put a null check that returns an empty string.